README.md

```diff
@@ -47,3 +40,95 @@ sudo ./check-payload scan operator \
 IMAGE=some.registry.location/check-payload
 podman  run --privileged -ti -v /:/myroot $IMAGE scan node --root /myroot
 ```
+
+## How it works
+
+`check-payload` gathers container images from OpenShift release payloads or
+container images. The scanner can be used on a RHEL or RHCOS node to scan the
+root image. A node scan runs within a container with the host OS mounted within
+it.
+
+### Container Image and Payload Scans
+
+Container Image and Payload Scans gather images and emit file paths to the
+validation logic. This is done by `podman` mounting the image to the local
+machine, walking the directory tree, and emitting the paths for executables to
+the validation engine.
+
+### Node Scans
+
+RHEL or RHCOS nodes can be scanned with `check-payload scan node`. To gather the
+file input paths the scanner queries for all the RPMs on the system and walks
+the paths within the RPMs finding executables. The list of executable paths are
+then processed by the validation engine.
+
+### Diagram
+
+```mermaid
+flowchart LR
+  subgraph inputs
+  Image --> Scanner
+  Payload --> Scanner
+  Node --> Scanner
+  end
+  Scanner --> Validations
+  Validations --> Printer
+  subgraph results
+  Printer
+  end
+  
+```
+
+### Validations
+
+The validation engine scans executables and golang executables independently. The scanner only scans for ELF executables.
+
+#### All
+
+All scans validate the inclusion of OpenSSL via libcrypto found in `/usr/lib64`
+or `/usr/lib`. The OpenSSL library is also validated to include `{FIPS_mode,
+fips_mode, or EVP_default_properties_is_fips_enabled}`.
+
+#### Regular Executables
+
+The rules to scan regular executables are:
+
+1. Must be dynamically link
+
+Most of RHEL/RHCOS executables are built dynamically to allow for dynamic
+linking to OpenSSL. There are exceptions for rule (1) which consists of some
+binaries (ldconfig, build-locale-archive, etc) which are required to be built
+statically, and/or do not provide cryptographic functionality.
+
+#### Golang
+
+Golang validations run through a pipeline:
+
+1. validateGoVersion
+   * Runs `go version -m` on the binary to cache the golang compiler version and details from the compile
+2. validateGoCgo
+   * Golang <= 1.17 skips
+   * validates the binary contains CGO_ENABLED=1 within the detailed output from validateGoVersion
+3. validateGoCGOInit
+   * Reads the binary and makes sure `cgo_init` is contained within (captures `x_cgo_init` as well)
+4. validateGoSymbols
+   * reads the golang symbol table
+   * If zero crypto symbols are found then skips
+   * Golang < 1.18 skips
```

Comment:
Not related to this PR, but maybe this should be the first step?

Reply:
Add a commit to change that (https://github.com/kolyshkin/check-payload/tree/go-sym-skip-early) but can't open a PR at the moment.